pycqed/instrument_drivers/meta_instrument/device.py

```diff
@@ -0,0 +1,1012 @@
+                                 (len(cal_states) != 0 and not classified) else None,
+                             'data_to_fit': {qbc.name: 'pf', qbt.name: 'pe'},
+                             'hard_sweep_params': hard_sweep_params,
+                             'soft_sweep_params': soft_sweep_params})
+        MC.run_2D(label, exp_metadata=exp_metadata)
+        if analyze:
+            if classified:
+                channel_map = {qb.name: [vn + ' ' +
+                                         qb.instr_uhf() for vn in
+                                         qb.int_avg_classif_det.value_names]
+                               for qb in [qbc, qbt]}
+            else:
+                channel_map = {qb.name: [vn + ' ' +
+                                         qb.instr_uhf() for vn in
+                                         qb.int_avg_det.value_names]
+                               for qb in [qbc, qbt]}
+            flux_pulse_tdma = tda.CPhaseLeakageAnalysis(
+                qb_names=[qbc.name, qbt.name],
+                options_dict={'TwoD': True, 'plot_all_traces': plot_all_traces,
+                              'plot_all_probs': plot_all_probs,
+                              'channel_map': channel_map})
+            cphases = flux_pulse_tdma.proc_data_dict[
+                'analysis_params_dict']['cphase']['val']
+            population_losses = flux_pulse_tdma.proc_data_dict[
+                'analysis_params_dict']['population_loss']['val']
+            leakage = flux_pulse_tdma.proc_data_dict[
+                'analysis_params_dict']['leakage']['val']
+            return cphases, population_losses, leakage, flux_pulse_tdma
+        else:
+            return
+
+    def measure_dynamic_phases(self, qbc, qbt, cz_pulse_name, hard_sweep_params=None,
+                               qubits_to_measure=None, cal_points=True,
+                               analyze=True, upload=True, n_cal_points_per_state=1,
+                               cal_states='auto', prep_params=None,
+                               exp_metadata=None, classified=False, update=False,
+                               reset_phases_before_measurement=True,
+                               prepend_n_cz=0):
+
+        if qubits_to_measure is None:
+            qubits_to_measure = [qbc, qbt]
+        if hard_sweep_params is None:
+            hard_sweep_params = {
+                'phase': {
+                    'values': np.tile(np.linspace(0, 2 * np.pi, 6) * 180 / np.pi, 2),
+                    'unit': 'deg'}}
+
+        if reset_phases_before_measurement:
+            dyn_phases = {qb.name: 0 for qb in qubits_to_measure}
+            self.get_pulse_par(cz_pulse_name,
+                               qbc, qbt, 'basis_rotation')(dyn_phases)
+        else:
+            dyn_phases = self.get_pulse_par(cz_pulse_name,
+                                            qbc, qbt, 'basis_rotation')()
+
+        qbc_name = qbc.name
+        qbt_name = qbt.name
```

Comment:
delete these two unused variables

Reply:
Addressed in https://github.com/QudevETH/PycQED_py3/pull/62/commits/90cdd02cd058de89758a207786f37740edd93bbf